vulture/core.py

```diff
@@ -351,9 +354,11 @@ def _handle_conditional_node(self, node, name):
 
     def _define(self, collection, name, first_node, last_node=None,
                 message='', confidence=DEFAULT_CONFIDENCE, ignore=None):
+        def _ignore_name(name):
+            return _match(name, self.ignore_names)
         last_node = last_node or first_node
         typ = collection.typ
-        if (ignore and ignore(self.filename, name)) or self._ignore_name(name):
+        if (ignore and ignore(self.filename, name)) or _ignore_name(name):
```

Comment:
`_ignore_name(name)` -> `_match(name, self.ignore_names)`

We also don't need `_ignore_decorator()`.